fuel-vm/src/interpreter/internal.rs

```diff
@@ -192,9 +187,8 @@ impl<S, Tx> Interpreter<S, Tx> {
     }
 
     pub(crate) fn set_frame_pointer(&mut self, fp: Word) {
-        self.context.update_from_frame_pointer(fp);
-
-        self.registers[RegId::FP] = fp;
+        let ReadRegisters { fp: register, .. } = split_registers(&mut self.registers).0;
```

Comment:
Hmm, we want to split it again. Maybe the compiler will see that we need access only to one register here and will optimize it, but I'm not sure=)

Maybe it is better to store registers like `ReadRegisters` and `Others`.

Reply:
I have benchmarked and it does optimize